code/game/machinery/teleporter.dm

```diff
@@ -192,25 +198,47 @@
 
 
 /obj/machinery/teleport/hub
-	name = "teleporter hub"
-	desc = "The teleporter hub handles all of the impossibly complex busywork required in instant matter transmission."
-	icon_state = "tele0"
-	dir = EAST
+	name = "teleporter pad"
+	desc = "The teleporter pad handles all of the impossibly complex busywork required in instant matter transmission."
+	icon_state = "pad"
 	idle_power_usage = 10
 	active_power_usage = 2000
 	var/obj/machinery/computer/teleporter/com
+	light_color = "#02d1c7"
+
+/obj/machinery/teleport/hub/Initialize()
+	. = ..()
+	queue_icon_update()
+
+/obj/machinery/teleport/hub/on_update_icon()
+	overlays.Cut()
+	if (com.station.engaged)
+		var/image/I = image(icon, src, "[initial(icon_state)]_active_overlay")
+		I.plane = EFFECTS_ABOVE_LIGHTING_PLANE
+		I.layer = ABOVE_LIGHTING_LAYER
+		overlays += I
+		set_light(0.4, 1.2, 4, 10)
+	else
+		set_light(0)
+		if (operable())
+			var/image/I = image(icon, src, "[initial(icon_state)]_idle_overlay")
+			I.plane = EFFECTS_ABOVE_LIGHTING_PLANE
+			I.layer = ABOVE_LIGHTING_LAYER
+			overlays += I
 
 /obj/machinery/teleport/hub/Bumped(var/atom/movable/M)
 	spawn()
```

Comment:
Possibly use a callback or remove this spawn? Unclear what it's here for.

Reply:
I have no idea what the spawn was for and didn't want to touch it for fear of it breaking something.